### PR TITLE
new approach for trust

### PR DESCRIPTION
**Pull request by AlvaroVega**
related https://github.com/telefonicasc/urbo-deployer/issues/868#issuecomment-1727080468
- [x] Added new config file: configTrust.js
- [x] Review doc about config.authentication and configTrust

### CODE REVIEW

**Comment by AlvaroVega** [comment]
New configuration has beed added into config.js file (remember that should be overwritten with a volume in a docker deploy):

```json
config.trusts = [
    {
        id: 'trust1',
        user: 'user1',
        password: 'password',
        service: 'domain1'
    },
    {
        id: 'trust2',
        user: 'user2',
        password: 'password2',
        service: 'domain2'
    }
];
```

**Comment by fgalan** [comment]
> New configuration has beed added into config.js file (remember that should be overwritten with a volume in a docker deploy):

As alternatives:

* Use a separate file, e.g. config.trust.js
* Use env var configuration, eg. `token1=user1,pass1,service1;token2=user2,pass2,service2;...` (similar approach to [per-service queue configuration in Orion](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/admin/perf_tuning.md#notification-modes-and-performance)) although if the number of tokens is large is a bit ugly.

File: docs/API/plain_rules.md [hunk 1]
```diff
@@ -396,23 +396,7 @@ the Perseo configuration). The `parameters` map includes the following fields:
...
-    ```json
-    "authentication": {
-        "host": "ext-keystone",
-        "port": 5001,
-        "user": "mycepuser",
-        "password": "myceppassword",
-        "service": "mycepuserservice"
-     }
-    ```
+    behind a PEP.
```
Comment (fgalan) [inline docs/API/plain_rules.md:399]: This modification means that `authentication` field is no longer supported?

In addition, the usage of the new configTrust.js file should be documented (although not sure in which .md file...)

Comment (AlvaroVega) [reply docs/API/plain_rules.md:399]: Witch this changes authentication is used partially, just host and port.
configTrust could be documented in a newAproachTrust.md ?

Comment (AlvaroVega) [reply docs/API/plain_rules.md]: doc was updated by https://github.com/telefonicaid/perseo-fe/pull/745/commits/fd58e1862cb7486f83a4708debcacbc490c88f06

**Comment by fgalan** [comment]
I think this section should be removed https://github.com/telefonicaid/perseo-fe/blob/11950e381cd7b924fcedff59a76e0babf8f81fb4/docs/architecture/architecture.md#authorization-server

**Comment by fgalan** [comment]
Also some env vars should be removed in configuration.md

| PERSEO_AUTHENTICATION_HOST         | Host of the authentication endpoint                                                                                                                                                                     |
| PERSEO_AUTHENTICATION_PORT         | Port of the authentication endpoint                                                                                                                                                                     |
| PERSEO_AUTHENTICATION_USER         | User to perform authentication                                                                                                                                                                          |
| PERSEO_AUTHENTICATION_PASSWORD     | Password for the user to perform authentication                                                                                                                                                         |
| PERSEO_AUTHENTICATION_SERVICE      | Keystone User Service to perform authentication

**Comment by AlvaroVega** [comment]
> Also some env vars should be removed in configuration.md
> 
> | PERSEO_AUTHENTICATION_HOST | Host of the authentication endpoint | | PERSEO_AUTHENTICATION_PORT | Port of the authentication endpoint | | PERSEO_AUTHENTICATION_USER | User to perform authentication | | PERSEO_AUTHENTICATION_PASSWORD | Password for the user to perform authentication | | PERSEO_AUTHENTICATION_SERVICE | Keystone User Service to perform authentication

done in f076d0043996b6490939f5dfcbf2a6872c4c3772

**Comment by fgalan** [comment]
> done in f076d0043996b6490939f5dfcbf2a6872c4c3772

I don't find that commit... maybe a `git push origin` is needed?

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -1,3 +1,5 @@
+- Add: new approach to handle trust auth (urbo-deployer#868)
+- Remove: field "authentication" support in update action
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:2]:
```suggestion
- Remove: field "authentication" support in update action
- Remove: config.authentication (along with related env vars PERSEO_AUTHENTICATION_*)
```

File: docs/admin/configuration.md [hunk 3]
```diff
@@ -107,10 +104,14 @@ Options for HA:
 
 Options for Authentication through PEP (for update action)
```
Comment (fgalan) [inline docs/admin/configuration.md:105]:
```suggestion
Options for Authentication through PEP (for update action) in the configTrust.js file:
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**